subiquity/server/controllers/filesystem.py

```diff
@@ -292,6 +297,97 @@ async def has_bitlocker_GET(self) -> List[Disk]:
                     break
         return [labels.for_client(disk) for disk in bitlockered_disks]
 
+    def get_partition(self, disk, number):
+        for p in disk.partitions():
+            if p._number == number:
+                return p
+        raise ValueError(f'Partition {number} on {disk.id} not found')
+
+    async def v2_GET(self) -> StorageResponseV2:
+        disks = self.model._all(type='disk')
+        need_root = not self.model.is_root_mounted()
+        need_boot = self.model.needs_bootloader_partition()
+        return StorageResponseV2(
+                disks=[labels.for_client(d) for d in disks],
+                todos_needs_root_mounted=need_root,
+                todos_needs_bootloader_partition=need_boot,
```

Comment:
I'd be fine with `need_root` and `need_boot` in `StorageResponseV2` too, to avoid so long names. dartfmt insists wrapping at 80 chars, and `todos_needs_bootloader_partition` is almost half of it.